File: backend/app/schemas/category.py
from uuid import UUID
from typing import Optional
from pydantic import BaseModel

class CategoryBase(BaseModel):
    name: str
    color_hex: str
    priority_level: int
    is_default: bool = False

class CategoryCreate(CategoryBase):
    pass

class Category(CategoryBase):
    id: UUID

    class Config:
        from_attributes = True

File: mobile/lib/features/events/data/models/category.dart
class Category {
  final String id;
  final String name;
  final String colorHex;
  final int priorityLevel;
  final bool isDefault;

  Category({
    required this.id,
    required this.name,
    required this.colorHex,
    this.priorityLevel = 5,
    this.isDefault = false,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as String,
      name: json['name'] as String,
      colorHex: json['color_hex'] as String,
      priorityLevel: json['priority_level'] as int? ?? 5,
      isDefault: json['is_default'] as bool? ?? false,
    );
  }
}
